File: sprint2/By languages/C#/lesson6code1.cs
using System;

public class Node {
    public Node Next;
    public int Value;
    
    public Node(int value) {
        Next = null;
        Value = value;
    }
}

public class LinkedList {
    public static void PrintLinkedList(Node node) {
        while (node != null) {
            Console.Write(node.Value + " -> ");
            node = node.Next;
        }
        Console.Write("None\n");
    }

    public static Node GetNodeByIndex(Node node, int index) {
        while (index > 0) {
            node = node.Next;
            index--;
        }
        return node;
    }

    public static Node InsertNode(Node head, int index, int value) {
        Node newNode = new Node(value);
        if (index == 0) {
            newNode.Next = head;
            return newNode;
        }
        Node previousNode = GetNodeByIndex(head, index-1);
        newNode.Next = previousNode.Next;
        previousNode.Next = newNode;
        return head;
    }
}

class Program {
    static void Main(string[] args) {
        Node n3 = new Node(3);
        Node n2 = new Node(2);
        Node n1 = new Node(1);
        n1.Next = n2;
        n2.Next = n3;
        Node node = n1;
        int index = 2;
        int value = 4;
        Node head = LinkedList.InsertNode(node, index, value);
        LinkedList.PrintLinkedList(head);
    }
}

File: sprint2/By languages/python/lesson6code1.py
def get_node_by_index(node, index):
		while index:
				node = node.next
				index -= 1
		return node

def insert_node(head, index, value):
		new_node = Node(value)
		if index == 0:
				new_node.next = head
				return new_node
		previous_node = get_node_by_index(head, index-1)
		new_node.next = previous_node.next 
		previous_node.next = new_node
		return head
>>> node, index, value = n1, 2, 'new_node'
>>> head = insert_node(node, index, value)
>>> print_linked_list(head)
first -> second -> new_node -> third -> None
